### ISSUE TITLE
Upcoming leapsecond updates for older releases

### ISSUE THREAD

**Issue by dlakaplan** [astropy/astropy#3791]
There will be a new leap second on 2015-06-30.  How will astropy deal with this?  If I understand correctly, for new releases/pulls from git the newer IERS tables will be included.  Is that correct?  If so, any idea when (the IERS A file already includes the new leap second)?

More importantly, is there a way to properly update those files for older releases? Some people hesitate to upgrade critical installations (used by large processing efforts) but we don't want times to get out of sync.  I see:

http://astropy.readthedocs.org/en/latest/time/#transformation-offsets

But that doesn't seem to store the updated info.   Is it as simple as (for IERS B):

```
from astropy.utils.iers import IERS_B, IERS_B_URL
from astropy.utils.data import download_file
import os
iers_b_file = download_file(IERS_B_URL, cache=True)
os.rename(iers_b_file, astropy.utils.iers.IERS_B_FILE)
```

(see also #3275) executed with appropriate permissions?  Can this be included as an appropriately bundled script?

**Comment by dlakaplan** [comment]
Sorry, I meant:

```
from astropy.utils.iers import IERS_B, IERS_B_URL, IERS_B_FILE
from astropy.utils.data import download_file
import os
iers_b_file = download_file(IERS_B_URL, cache=True)
os.rename(iers_b_file, IERS_B_FILE)
```

**Comment by mhvk** [comment]
@dlakaplan - note that the IERS tables only affect UTC->UT1, not the leap seconds in, e.g., UTC->TAI. It does mean that both IERS and ERFA(?) needs to be updated.  CC @taldcroft.  And cc @eteq as the one who thought he could make a nice automatic script. Also see #3275.

**Comment by taldcroft** [comment]
The current master for the ERFA project has the leap second in it (thanks to @timj with https://github.com/liberfa/erfa/pull/26).

So we need:
- A release of ERFA (v1.2?)
- Incorporate the ERFA 1.2 into astropy.
- A bugfix release of astropy.

Now here is where it gets interesting.  We want to get this into an astropy release soon, and definitely well before June 30.  That means a bugfix / patch release.  I *think* that the update to ERFA can be mostly classified (at least for our purposes) as a bugfix release, though it does appear some "duplicate" functions were removed.  I haven't dug into exactly what that means. 

@eteq @timj - thoughts?

As @dlakaplan notes it would be nice to backport the leap second update to at least 0.4.x.  Can that be done simply by a change to `dat.c` like [here](https://github.com/liberfa/erfa/pull/26/files#diff-94301bf2bcc3883edb8261e25028d9eeR191)?

**Comment by taldcroft** [comment]
Oops, I meant to cc @embray on this for comment about a bugfix release.

**Comment by timj** [comment]
I think just copying the `dat.c` to an older release is all that is needed. That file doesn't care about any other part of ERFA. Regarding the duplication -- I think that was because a header file had declared the same function twice.

**Comment by embray** [comment]
I think in the past I've said that if we want to release leapsecond updates we can do it as patch releases.  I agree it should be backported to 0.4.x as well.

**Comment by eteq** [comment]
I think we could possibly have considered this closed by #3794, but definitely #3802 should do the job.

**Comment by embray** [comment]
Yup, this is ready to go.
